lmdeploy/pytorch/utils.py

```diff
@@ -1,135 +1,102 @@
 # Copyright (c) OpenMMLab. All rights reserved.
 
-import re
+import logging
 
-from transformers import (PreTrainedTokenizerFast, StoppingCriteria,
-                          StoppingCriteriaList)
 from transformers.generation.streamers import BaseStreamer
 
+from .dist import get_rank, master_only, master_only_and_broadcast_general
 
-def get_utils(model):
-    """Get utils by model type."""
+try:
+    import readline  # To support command line history # noqa: F401
+except ImportError:  # readline not available
+    pass
```

Comment:
This is as expected because `import readline` just takes effect. After importing this module, user can retrieve history input via arrow keys, during chat. So we don't need to really *use* it as normal python packages, but need to import it.